src/graphql/search/map/normalize.ts

```diff
@@ -0,0 +1,112 @@
+import queryString, { ParsedUrlQueryInput } from 'querystring'
+import { MAX_ZOOM, MIN_ZOOM, MapLayerTypes } from './config'
+import { MapCollection, MapCollectionLayer, LegendItemMapLayer } from '../../../generated/graphql'
+
+type InitialLegendItem = {
+  params: ParsedUrlQueryInput
+} & LegendItemMapLayer
+
+type InitialMapLayer = {
+  minZoom: number
+  maxZoom: number
+  params: ParsedUrlQueryInput
+  legendItems: Array<InitialLegendItem>
+} & MapCollectionLayer
+
+type InitialMapCollection = {
+  mapLayers: Array<InitialMapLayer>
+} & MapCollection
+
+export function composeId(collectionId: string, id: string) {
+  return `${collectionId}-${id}`
+}
+
+export function findMapLayer(mapLayers: Array<InitialMapLayer>, id: string) {
+  return mapLayers.find(mapLayer => mapLayer.id === id)
+}
+
+export function normalizeLegendItems(
+  collectionId: string,
+  legendItems: Array<InitialLegendItem>,
+  mapLayers: Array<InitialMapLayer>,
+) {
+  return legendItems.map(legendItem => {
+    const legendItemLayer = legendItem.id ? findMapLayer(mapLayers, legendItem.id) : null
+    const notSelectable = legendItem.notSelectable || !legendItem.id || false // legendItems with an id are always selectable, unless defined otherwise
```

Comment:
Since the last expression here is actually negated and thus cast to a boolean we can do without the `|| false` here.